File: core/src/main/java/com/example/forestcollect/game/world/GraphicsQuality.java
package com.example.forestcollect.game.world;

public enum GraphicsQuality {
    LOW("Низкое", 0.6f, 40),
    MEDIUM("Среднее", 0.9f, 65),
    HIGH("Высокое", 1.2f, 90);

    private final String displayName;
    private final float foliageDensity;
    private final int treeCount;

    GraphicsQuality(String displayName, float foliageDensity, int treeCount) {
        this.displayName = displayName;
        this.foliageDensity = foliageDensity;
        this.treeCount = treeCount;
    }

    public float getFoliageDensity() {
        return foliageDensity;
    }

    public int getTreeCount() {
        return treeCount;
    }

    public String getDisplayName() {
        return displayName;
    }

    @Override
    public String toString() {
        return displayName;
    }
}

File: app/src/main/java/com/forestcoins/game/data/GraphicsQuality.kt
package com.forestcoins.game.data

enum class GraphicsQuality(val treeCount: Int, val coinCount: Int, val renderDistance: Float) {
    LOW(treeCount = 40, coinCount = 30, renderDistance = 80f),
    MEDIUM(treeCount = 70, coinCount = 50, renderDistance = 120f),
    HIGH(treeCount = 100, coinCount = 70, renderDistance = 160f);

    companion object {
        fun fromKey(key: String): GraphicsQuality = entries.firstOrNull { it.name == key } ?: HIGH
    }
}
